src/main/java/org/zaproxy/zap/extension/jwt/fuzzer/messagelocations/JWTMessageLocation.java

```diff
@@ -0,0 +1,167 @@
+            String key,
+            boolean isHeaderOrPayload,
+            FuzzerJWTSignatureOperation fuzzerJWTSignatureOperation) {
+        super(location, start, end, value);
+        this.key = key;
+        this.isHeaderField = isHeaderOrPayload;
+        this.fuzzerJWTSignatureOperation = fuzzerJWTSignatureOperation;
+    }
+
+    /** @return JSON Object's Key field */
+    public String getKey() {
+        return key;
+    }
+
+    /** @param key JSON Object's Key field */
+    public void setKey(String key) {
+        this.key = key;
+    }
+
+    /** @return true of Key field is present in Header */
+    public boolean isHeaderField() {
+        return isHeaderField;
+    }
+
+    /** @param isHeaderField */
+    public void setHeaderField(boolean isHeaderField) {
+        this.isHeaderField = isHeaderField;
+    }
+
+    /** @return FuzzerJWTSignatureOperation */
+    public FuzzerJWTSignatureOperation getFuzzerJWTSignatureOperation() {
+        return fuzzerJWTSignatureOperation;
+    }
+
+    /** @param fuzzerJWTSignatureOperation */
+    public void setFuzzerJWTSignatureOperation(
+            FuzzerJWTSignatureOperation fuzzerJWTSignatureOperation) {
+        this.fuzzerJWTSignatureOperation = fuzzerJWTSignatureOperation;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = super.hashCode();
+        result =
+                prime * result
+                        + ((fuzzerJWTSignatureOperation == null)
+                                ? 0
+                                : fuzzerJWTSignatureOperation.hashCode());
+        result = prime * result + (isHeaderField ? 1231 : 1237);
+        result = prime * result + ((key == null) ? 0 : key.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (!super.equals(obj)) return false;
+        if (getClass() != obj.getClass()) return false;
+        JWTMessageLocation other = (JWTMessageLocation) obj;
+        if (fuzzerJWTSignatureOperation != other.fuzzerJWTSignatureOperation) return false;
+        if (isHeaderField != other.isHeaderField) return false;
+        if (key == null) {
+            if (other.key != null) return false;
+        } else if (!key.equals(other.key)) return false;
+        return true;
+    }
+
+    @Override
+    public int compareTo(MessageLocation otherLocation) {
+        if (!(otherLocation instanceof JWTMessageLocation)) {
+            return 1;
+        }
+        JWTMessageLocation that = (JWTMessageLocation) otherLocation;
+
+        int result = Boolean.compare(that.isHeaderField, this.isHeaderField);
+        if (result != 0) {
+            return result;
+        }
+
+        result = this.key.compareTo(that.key);
+        if (result != 0) {
+            return result;
+        }
+
+        result = this.fuzzerJWTSignatureOperation.compareTo(that.fuzzerJWTSignatureOperation);
```

Comment:
I think there is no benefit of adding signatureOperation as all the signatureOperations will be same.